extensions/notebook/src/book/bookTocManager.ts

```diff
@@ -354,10 +362,10 @@ export class BookTocManager implements IBookTocManager {
 	public async updateBook(element: BookTreeItem, targetBook: BookTreeItem, targetSection?: JupyterBookSection): Promise<void> {
 		const targetBookVersion = targetBook.book.version === BookVersion.v1 ? BookVersion.v1 : BookVersion.v2;
 		if (element.contextValue === 'section') {
-			await this.addSection(element, targetBook);
-			const elementVersion = element.book.version === BookVersion.v1 ? BookVersion.v1 : BookVersion.v2;
 			// modify the sourceBook toc and remove the section
 			const findSection: JupyterBookSection = { file: element.book.page.file?.replace(/\\/g, '/'), title: element.book.page.title };
+			await this.addSection(element, targetBook);
+			const elementVersion = element.book.version === BookVersion.v1 ? BookVersion.v1 : BookVersion.v2;
```

Comment:
This can only be v1 or v2 so not sure what this check is doing. Could just be this right?

`const elementVersion = element.book.version;`

Reply:
Yup, I just need to cast it so it's a BookVersion object.